File: problems/done/2126.py
def main():
    case = 1
    while True:
        try:
            substring = input()
            string = input()
        except EOFError:
            break

        print('Caso #{}:'.format(case))

        qtd = string.count(substring)
        last = string.rfind(substring)

        if qtd:
            print('Qtd.Subsequencias: {}'.format(qtd))
            print('Pos: {}'.format(last + 1))
        else:
            print('Nao existe subsequencia')
        case += 1
        print()

main()

File: problems/c/2126.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(void)
{
    int caso = 1, qtde;
    char subseq[100], seq[200], *pos, *ultima_pos;

    while (scanf("%s\n", subseq) != EOF)
    {
        scanf("%s", seq);

        pos = seq, ultima_pos = NULL, qtde = 0;

        while (1)
        {
            pos = strstr(pos, subseq);
            if (pos == NULL)
                break;
            ultima_pos = pos;
            qtde++, pos++;
        }

        printf("Caso #%d:\n", caso++);
        if (qtde)
        {
            printf("Qtd.Subsequencias: %d\n", qtde);
            printf("Pos: %ld\n", ultima_pos - seq + 1);
        }
        else
        {
            printf("Nao existe subsequencia\n");
        }
        putchar('\n');
    }

    return 0;
}
